packages/muelu/src/Graph/MueLu_CoalesceDropFactory_def.hpp

```diff
@@ -828,7 +828,7 @@ namespace MueLu {
       GetOStream(Statistics1) << "CoalesceDropFactory: nodeMap " << nodeMap->getNodeNumElements() << "/" << nodeMap->getGlobalNumElements() << " elements" << std::endl;
 
       // 3) create graph of amalgamated matrix
-      RCP<CrsGraph> crsGraph = CrsGraphFactory::Build(nodeMap, 10, Xpetra::DynamicProfile);
+      RCP<CrsGraph> crsGraph = CrsGraphFactory::Build(nodeMap, A->getNodeMaxNumRowEntries()*blockdim, Xpetra::StaticProfile);
```

Comment:
This code might run fine, but it could blow out memory if one row happens to have many more entries than other rows.  Once we make this change, we may also want to duplicate the loop below, once to get a number of entries in each row (it looks like the loop is amenable to that), and once again to fill the CrsGraph.

Reply:
@mhoemmen That's a good suggestion, I'll push an update.